ledger-service/http-json/src/bench/scala/com/daml/http/dbbackend/ContractDaoBenchmark.scala

```diff
@@ -107,5 +100,89 @@ abstract class ContractDaoBenchmark extends OracleAround {
       .unsafeRunSync()
     assert(inserted == batchSize)
   }
+}
+
+trait BenchmarkDbConnection {
+  def connectToDb(): Unit
+  def createDbJdbcConfig: JdbcConfig
+  def cleanup(): Unit
+}
+
+trait OracleBenchmarkDbConn extends BenchmarkDbConnection with OracleAround {
+
+  private var user: User = _
+  private val disableContractPayloadIndexing = false
+
+  override def connectToDb() = {
+    connectToOracle()
+  }
 
+  override def createDbJdbcConfig: JdbcConfig = {
+    user = createNewRandomUser()
+    val cfg = JdbcConfig(
+      dbutils.JdbcConfig(
+        driver = "oracle.jdbc.OracleDriver",
+        url = oracleJdbcUrl,
+        user = user.name,
+        password = user.pwd,
+        poolSize = ConnectionPool.PoolSize.Integration,
+      ),
+      dbStartupMode = DbStartupMode.CreateOnly,
+      backendSpecificConf =
+        if (disableContractPayloadIndexing) Map(DisableContractPayloadIndexing -> "true")
+        else Map.empty,
+    )
+    cfg
+  }
+
+  override def cleanup(): Unit = {
+    dropUser(user.name)
+  }
+}
+
+/*
+ Running the PG benchmark requires us to explicitly set the PGPASSWORD env variable which is picked
+ by the `createdb` tool when creating relevant database e.g
+
+ $ PGPASSWORD="password" bazel run //ledger-service/http-json:contractdao-bench -- -f 1 -i 1 -wi 1 -bm avgt \
+ "QueryPayloadBenchmarkPostgres" -p extraParties=1,10 -p extraPayloadValues=1,10,100 -p batchSize=1000
+
+ set the password accordingly
+ */
+trait PostgresBenchmarkDbConn extends BenchmarkDbConnection with PostgresAround {
+
+  @volatile
+  private var database: PostgresDatabase = _
+
+  private val host = sys.env.getOrElse("POSTGRESQL_HOST", "localhost")
+  private val port = Port(sys.env.get("POSTGRESQL_PORT").map(_.toInt).getOrElse(5432))
+  private val user = sys.env.getOrElse("POSTGRESQL_USERNAME", "postgres")
+  private val password = sys.env.getOrElse(
```

Comment:
the usage of `createdb` tool to create databases means that we have an unfortunate requirement for setting this env variable to make authentication work. Open to alternative, more intuitive suggestions here.